### PR TITLE
Move unit tests to TaskCluster

### LINKED ISSUE #2961: Move unit tests to TaskCluster so they can run when bumping version
Right now they depend on the pre-built decoder package from TC which is not available yet when the tests run.

### PR DESCRIPTION
**Pull request by imskr**
Fixes #2961

### CODE REVIEW

File: taskcluster/tc-train-unittests.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+mkdir -p /tmp/train || true
+mkdir -p /tmp/train_tflite || true
+
+virtualenv_activate "${pyalias}" "deepspeech"
+
+set -o pipefail
+pip install --upgrade pip==19.3.1 setuptools==45.0.0 wheel==0.33.6 | cat
+pushd ${HOME}/DeepSpeech/ds
+    pip install --upgrade . | cat
+popd
```
Comment (reuben) [inline taskcluster/tc-train-unittests.sh:21]: This is causing the shell to leave the clone directory, so the `python -m unittest` below doesn't find any tests. It should be moved inside this `pushd` block.

**Comment by reuben** [comment]
Oh, and you can also remove the Travis entry for running unittests.

**Comment by imskr** [comment]
Thanks, @reuben I'm on it.

File: taskcluster/tc-train-unittests.sh [hunk 2]
```diff
@@ -4,21 +4,14 @@ set -xe
 
 source $(dirname "$0")/tc-tests-utils.sh
 
-extract_python_versions "$1" "pyver" "pyver_pkg" "py_unicode_type" "pyconf" "pyalias"
```
Comment (reuben) [inline taskcluster/tc-train-unittests.sh]: Pretty sure you still need this, it's what defines the pyalias variable used below.

File: taskcluster/test-training-unittests_8k-linux-amd64-py35m-opt.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    dependencies:
+      - "linux-amd64-ctc-opt"
+    system_setup:
+      >
+        apt-get -qq update && apt-get -qq -y install ${training.packages_trusty.apt}
+    args:
+      tests_cmdline: "${system.homedir.linux}/DeepSpeech/ds/taskcluster/tc-train-unittests.sh 3.5.8:m"
+    metadata:
+      name: "DeepSpeech on Linux AMD64 CPU training unittests using Python 3.5"
+      description: "Training a DeepSpeech LDC93S1 model for Linux/AMD64 Python 3.5, CPU only, optimized version"
```
Comment (reuben) [inline taskcluster/test-training-unittests_8k-linux-amd64-py35m-opt.yml:12]: Can you also update the descriptions?

File: taskcluster/tc-train-unittests.sh (hunk 2 above)
Comment (imskr) [reply taskcluster/tc-train-unittests.sh]: ohh deleted it by mistake!

File: taskcluster/test-training-unittests_8k-linux-amd64-py35m-opt.yml (hunk 3 above)
Comment (imskr) [reply taskcluster/test-training-unittests_8k-linux-amd64-py35m-opt.yml:12]: What should I write there?

Comment (reuben) [reply taskcluster/test-training-unittests_8k-linux-amd64-py35m-opt.yml:12]: How about "Training unit tests for Linux/AMD64 Python 3.5, CPU only, optimized version" (with version adjusted for the other two, of course)

**Comment by imskr** [comment]
@reuben Now looks good?

**Comment by reuben** [comment]
PR looks good, but there's a weird failure on Python 3.5...

**Comment by imskr** [comment]
@reuben Yeah that's weird!

**Comment by reuben** [comment]
Closed in #3025

**Pull request merged**